sections/exchange/hooks/useBalancerExchange.tsx

```diff
@@ -185,13 +190,7 @@ const useBalancerExchange = ({
 		if (isApproving) {
 			return t('exchange.summary-info.button.submitting-approval');
 		}
-		if (
-			!isWalletConnected ||
-			baseCurrencyAmountBN.isNaN() ||
-			quoteCurrencyAmountBN.isNaN() ||
-			baseCurrencyAmountBN.lte(0) ||
-			quoteCurrencyAmountBN.lte(0)
-		) {
+		if (!isWalletConnected || baseCurrencyAmountBN.lte(0) || quoteCurrencyAmountBN.lte(0)) {
```

Comment:
same q about the logic not making sense?

Reply:
the logic is equivalent to what it was before here

Reply:
in the previous code, you get NaN from `toBigNumber` parsing a non-integer. in the new code, if the text is not a number, it will trigger an exception.

in the earlier code, it does not catch the exception, so I added that in a new commit.